libs/src/services/creatorNode/index.js

```diff
@@ -752,6 +779,77 @@ class CreatorNode {
     }
   }
 
+  async _uploadResumableTrackFile (file, onProgress) {
+    await this.ensureConnected()
+    const { headers } = this.createFormDataAndUploadHeaders(file)
+    const url = this.creatorNodeEndpoint + '/track_content_upload'
+
+    try {
+      return await this.startResumableUpload(file, url, headers, onProgress)
+    } catch (e) {
+      this._handleErrorHelper(e, url, headers['X-Request-ID'])
+    }
+  }
+
+  // https://github.com/tus/tus-js-client/blob/master/docs/api.md
+  async startResumableUpload (file, url, headers, onProgress) {
+    const randomFileName = uuid()
+    return new Promise((resolve, reject) => {
+      const upload = new tus.Upload(file, {
+        // Endpoint is the upload creation URL from your tus server
+        endpoint: url,
+        // Retry delays will enable tus-js-client to automatically retry on errors
+        retryDelays: [0, 3000, 5000, 10000, 20000],
+        // Attach additional metadata about the file for the server
+        headers: {
+          ...headers,
+          filesize: file.size,
+          filetype: file.type,
+          filename: file.name,
+          randomfilename: randomFileName // does not have extension
```

Comment:
since this isn't used in the fingerprinting and if this isn't needed client side can we just generate this in the route?

Reply:
+1 this feels like an attack vector. someone could put things here to cause collisions